### PR TITLE
Add monoidal & enriched categories

### LINKED ISSUE #643: Make `_⋆_`, `_∘_` right associative in `Category`
So that we can write
```
f ⋆ g ⋆ h
```
and Agda will know how to parse it.

It's as easy as putting
```agda
  infixr 9 _⋆_
  infixr 9 _∘_
```
at the end of the definition of `Category`.

We could also think about doing this for other associative algebraic structures (semigroups, groups, rings, ...)

### PR DESCRIPTION
**Pull request by barrettj12**
This PR depends on #654, so that will have to be merged first.

Will close #643.

### CODE REVIEW

File: Cubical/Categories/Monoidal/Base.agda [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+
+  record MonoidalStr : Type (ℓ-max ℓ ℓ') where
+    field
+      tenstr : TensorStr
+
+    open TensorStr tenstr public
+
+    private
+      -- Give some names to things
```
Comment (mortberg) [inline Cubical/Categories/Monoidal/Base.agda:55]: These names are not great... Do they get exported somehow or are they just local?

File: Cubical/Categories/Monoidal/Base.agda [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+
+module _ {ℓ ℓ' : Level} (C : Category ℓ ℓ') where
+  open Category C
+
+  private
+    record TensorStr : Type (ℓ-max ℓ ℓ') where
+      field
+        ─⊗─ : Functor (C × C) C
+        i : ob
```
Comment (mortberg) [inline Cubical/Categories/Monoidal/Base.agda:22]: `i` is very short... Maybe a slightly more descriptive name would be good?

**Review by mortberg — CHANGES_REQUESTED** [review]
Looks mostly good to me, just two small comments/questions on naming. Note that there is now a conflict that needs to be fixed before merging

File: Cubical/Categories/Monoidal/Base.agda (hunk 2 above)
Comment (barrettj12) [reply Cubical/Categories/Monoidal/Base.agda:22]: `unit`?

Comment (mortberg) [reply Cubical/Categories/Monoidal/Base.agda:22]: Much better!

File: Cubical/Categories/Monoidal/Base.agda (hunk 1 above)
Comment (barrettj12) [reply Cubical/Categories/Monoidal/Base.agda]: Nope, I checked and they are not exported. They are just there to make the "axioms" below look nice (like equations).

**Comment by mortberg** [comment]
There is still a conflict

**Review by mortberg — APPROVED** [review]

**Comment by barrettj12** [comment]
> There is still a conflict

Whoops forgot, fixed now.

**Comment by mortberg** [comment]
> > There is still a conflict
> 
> Whoops forgot, fixed now.

Great! Merging once CI is done

**Comment by barrettj12** [comment]
Also this will close #643

**Pull request merged**